src/Export/ProductService.php

```diff
@@ -261,4 +244,50 @@ protected function addProductIdFilters(Criteria $criteria, string $productId): v
             )
         );
     }
+
+    protected function getVisibleProducts(?int $limit, ?int $offset, ?string $productId): EntitySearchResult
+    {
+        $criteria = $this->getCriteriaWithProductVisibility($limit, $offset);
+
+        if ($productId) {
+            $this->addProductIdFilters($criteria, $productId);
+        }
+
+        return $this->container->get('product.repository')->search(
+            $criteria,
+            $this->salesChannelContext->getContext()
+        );
+    }
+
+    protected function buildProductsWithVariantInformation(EntitySearchResult $result): ProductCollection
+    {
+        $products = new ProductCollection();
+
+        /** @var ProductEntity $product */
+        foreach ($result->getEntities() as $product) {
+            if ($product->getMainVariantId() !== null) {
+                $product = $this->getRealMainProductWithVariants($product->getMainVariantId());
+            } else {
+                $this->assignChildrenOrSiblings($product);
+            }
```

Comment:
if ($product->getMainVariantId() !== null) {
                $product = $this->getRealMainProductWithVariants($product->getMainVariantId());
            }
            $this->assignChildrenOrSiblings($product);